Create 2025-09-17-update-preview-url-setting.mdx (#25236)
cloudflare/cloudflare-docs: src/content/changelog/workers/2025-09-17-update-preview-url-setting.mdx

---
title: Updates to Workers Preview URL Settings
description: We performed a one-time update to disable preview URLs on Workers with workers.dev disabled to require an explicit opt-in. 
date: 2025-09-17
---
 

> To prevent the accidental exposure of applications, we've updated how Worker preview URLs (<WORKER_NAME>.<SUBDOMAIN>.workers.dev) are handled.

Reviewer: ```suggestion
To prevent the accidental exposure of applications, we've updated how Worker preview URLs (<PREVIEW>-<WORKER_NAME>.<SUBDOMAIN>.workers.dev) are handled.
```

**What has changed:**

> - New Default Behavior: If a Worker's `workers.dev` route is disabled and its preview URL is not explicitly configured, the preview URL will now default to disabled. However, you can still explicitly enable the preview URL regardless of your `workers.dev` enablement status. 

Reviewer: ```suggestion
- New Default Behavior: If a Worker's `workers.dev` route is disabled and its preview URL is not explicitly configured, the preview URL will now default to disabled. However, you can still explicitly enable the preview URL regardless of your `workers.dev` enablement status by doing `preview_urls = true`
```

> - New Default Behavior: If a Worker's [`workers.dev`](/workers/configuration/routing/workers-dev/) route is disabled and its preview URL is not explicitly configured, the preview URL will now default to disabled. However, you can still explicitly enable the preview URL regardless of your `workers.dev` enablement status by doing `preview_urls = true`

Reviewer 2: ```suggestion
- New Default Behavior: If a Worker's [`workers.dev`](/workers/configuration/routing/workers-dev/) route is disabled and its preview URL is not explicitly configured, the preview URL will now default to disabled. However, you can still explicitly enable the preview URL regardless of your `workers.dev` enablement status by configuring it in the dashboard, or by setting `preview_urls = true` in your Wrangler configuration file.
```

- One-Time Update for Existing Workers: We performed a one-time update to disable preview URLs on existing Workers with a disabled `workers.dev` route. This ensures an explicit opt-in is required going forward.

> - Re-enabling Preview URLs: If your preview URL was disabled, you can re-enable it through the Worker's settings page in the dashboard or by updating the Worker's configuration file and deploying with the latest version of Wrangler. 

Reviewer: is there a min ver we can specify?
`of Wrangler (minimum of v4.xx.xx)`

> - Re-enabling Preview URLs: If your preview URL was disabled, you can re-enable it through the Worker's settings page in the dashboard or by updating the Worker's configuration file and deploying with Wrangler (minimum of version 4.34.0).

Reviewer 2: How?

> - Re-enabling Preview URLs: If your preview URL was disabled, you can re-enable it through the Worker's settings page in the dashboard or by updating the Worker's configuration file and deploying with Wrangler (minimum of version 4.34.0).

Reviewer 3: The problem here is that older Wrangler versions **default** to `preview_urls = true`. So from the API, we don't know if it was explicitly enabled by the user, or just the default. Due to this, we made `4.34.0` default to `preview_urls = false`, so whenever we receive `true`, we'll know it was the explicit intent of the user.

We performed a one-time update to disable preview URLs for existing Workers where the [workers.dev subdomain](/workers/configuration/routing/workers-dev/) was also disabled.

Because preview URLs were historically enabled by default, users who had intentionally disabled their workers.dev route may not have realized their Worker was still accessible at a separate preview URL. This update was performed to ensure that using a preview URL is always an intentional, opt-in choice.

If your Worker was affected, its preview URL (`<PREVIEW>-<WORKER_NAME>.<SUBDOMAIN>.workers.dev`) will now direct to an informational page explaining this change.

**How to Re-enable Your Preview URL**

If your preview URL was disabled, you can re-enable it [via the Cloudflare dashboard](/workers/configuration/previews/#toggle-preview-urls-enable-or-disable) by navigating to your Worker's Settings page and toggling on the Preview URL.

Alternatively, you can use Wrangler by adding the `preview_urls = true` setting to your Wrangler file and redeploying the Worker.

<WranglerConfig>
```jsonc title="wrangler.jsonc"
{
  "preview_urls": true
}
```
</WranglerConfig>

> Note: You can set `preview_urls = true` with any Wrangler version that supports the preview URL flag (v3.9.0+). However, we recommend updating to v4.34.0 or newer, as this version defaults `preview_urls` to false, ensuring preview URLs are always enabled by explicit choice.

Reviewer 2: ```suggestion
Note: You can set `preview_urls = true` with any Wrangler version that supports the preview URL flag (v3.91.0+). However, we recommend updating to v4.34.0 or newer, as this version defaults `preview_urls` to false, ensuring preview URLs are always enabled by explicit choice.
```

If your Worker was affected, its preview URL (`<PREVIEW>-<WORKER_NAME>.<SUBDOMAIN>.workers.dev`) will now direct to an informational page explaining this change.

**How to Re-enable Your Preview URL**

If your preview URL was disabled, you can re-enable it [via the Cloudflare dashboard](/workers/configuration/previews/#toggle-preview-urls-enable-or-disable) by navigating to your Worker's Settings page and toggling on the Preview URL.

Alternatively, you can use Wrangler by adding the `preview_urls = true` setting to your Wrangler file and redeploying the Worker.

<WranglerConfig>
```jsonc title="wrangler.jsonc"
{
  "preview_urls": true
}
```
</WranglerConfig>

Note: You can set `preview_urls = true` with any Wrangler version that supports the preview URL flag (v3.91.0+). However, we recommend updating to v4.34.0 or newer, as this version defaults `preview_urls` to false, ensuring preview URLs are always enabled by explicit choice.

> ## New Default Behavior Going Forward

Reviewer 4: This feels extraneous - should be part of the first paragraph. It’s an overly wordy header.

Going forward, if a Worker's `workers.dev` subdomain is disabled, its preview URL will be disabled by default.

> To activate the preview URL, you must explicitly enable it using one of the methods described above. 

Reviewer 4: We say this earlier. This whole section should be removed.

Author: updated to remove it
